lib/cartopy/mpl/geoaxes.py

```diff
@@ -1674,23 +1674,41 @@ def _pcolormesh_patched(self, *args, **kwargs):
             wrap_proj_types = (ccrs._RectangularProjection,
                                ccrs._WarpedRectangularProjection,
                                ccrs.InterruptedGoodeHomolosine,
-                               ccrs.Mercator)
+                               ccrs.Mercator,
+                               ccrs.Stereographic)
             if isinstance(t, wrap_proj_types) and \
                     isinstance(self.projection, wrap_proj_types):
 
                 C = C.reshape((Ny - 1, Nx - 1))
                 transformed_pts = transformed_pts.reshape((Ny, Nx, 2))
 
-                # Compute the length of edges in transformed coordinates
+                # Compute the length of diagonals in transformed coordinates
+                # This takes into account the case where the top of the cell
+                # is on one side of the boundary while the other is on the
+                # other side, case overlooked with the edges only.
+                #
+                #    Top----Top  |
+                #                |  Bottom-----Bottom
+                #
                 with np.errstate(invalid='ignore'):
-                    edge_lengths = np.hypot(
-                        np.diff(transformed_pts[..., 0], axis=1),
-                        np.diff(transformed_pts[..., 1], axis=1)
+                    ptx, pty = transformed_pts[..., 0], transformed_pts[..., 1]
+                    diagonal0_lengths = np.hypot(
+                        ptx[1:, 1:] - ptx[:-1, :-1],
+                        pty[1:, 1:] - pty[:-1, :-1]
+                    )
+                    diagonal1_lengths = np.hypot(
+                        ptx[1:, :-1] - ptx[:-1, 1:],
+                        pty[1:, :-1] - pty[:-1, 1:]
                     )
                     to_mask = (
-                        (edge_lengths > abs(self.projection.x_limits[1] -
-                                            self.projection.x_limits[0]) / 2) |
-                        np.isnan(edge_lengths)
+                        (diagonal0_lengths > (
+                            abs(self.projection.x_limits[1]
+                                - self.projection.x_limits[0])) / 2) |
```

Comment:
I am confused why the check is against (xlim1 - xlim0)/2 here. What if I wanted to plot a single cell mesh that is from -170 to 170? I think this would actually mask that box, even though there is no wrapping going on.

I'm wondering if a strictly increasing/decreasing check would be able to do something similar or add that to this.

Reply:
This check is a standard check to remove the wrapping. 
But of course it has its drawback, it is not perfect. It would mask a grid with one big cell.
I would have to check, but I am not sure that a one cell grid would go in that part, but obviously if one designs a 3 x 3 grid with one cell bigger than half the plot, the big cell would be hidden. But I would think this is far from a standard case.

As we cannot have 2 cells bigger than half the plot, I would suggest to check or to fix the case where the grid has only one cell in one direction, later.

Now if the grid has only one big cell spanning over more than 180 degree how do you know which way around the earth it should be plotted? Does it go from -170 to 170 or more probably from -190 to -170? 
Please see the comment I made on #522 with examples on how it is impossible to know how a unique big cell should be plotted. 

NOTE, that there is an other fix on the same subject as currently the hidden cells are still plotted a little further in the code.

Reply:
I agree that there is quite a bit of ambiguity here. I still think this addition is a good improvement to capture another "edge" case right now.